pyannote/database/custom.py

```diff
@@ -338,27 +441,38 @@ def create_protocol(
             warnings.warn(msg)
             continue
 
-        method_name = f"{subset}_iter"
-
-        if database == "X":
-            methods[method_name] = functools.partial(
-                meta_subset_iter,
-                database,
-                task,
-                protocol,
-                subset,
-                subset_entries,
-                database_yml,
-            )
+        
+        if 'trial' in subset_entries.keys():
+            methods[f"{subset}_trial"] = functools.partialmethod(
+                    subset_trial,
+                    database=database,
+                    task=task,
+                    protocol=protocol,
+                    subset=subset,
+                    entries=subset_entries,
+                    database_yml=database_yml,
+                )
         else:
-            methods[method_name] = functools.partialmethod(
-                subset_iter,
-                database=database,
-                task=task,
-                protocol=protocol,
-                subset=subset,
-                entries=subset_entries,
-                database_yml=database_yml,
+            method_name = f"{subset}_iter"
+            if database == "X":
+                methods[method_name] = functools.partial(
+                    meta_subset_iter,
+                    database,
+                    task,
+                    protocol,
+                    subset,
+                    subset_entries,
+                    database_yml,
+                )
+            else:
+                methods[method_name] = functools.partialmethod(
+                    subset_iter,
+                    database=database,
+                    task=task,
+                    protocol=protocol,
+                    subset=subset,
+                    entries=subset_entries,
+                    database_yml=database_yml,
             )
```

Comment:
When `trial` key exists, we should define both `{subset}_iter` and `{subset}_trial` because `SpeakerVerificationProtocol` is a subclass of `SpeakerDiarizationProtocol`.

```python
if database == "X":
    # define meta_subset_iter here
else:
    # define subset_iter here
    if 'trial' in subset_entries:
        # define subset_trial here
```